src/vm/jitinterface.cpp

```diff
@@ -6991,7 +6991,7 @@ bool getILIntrinsicImplementationForUnsafe(MethodDesc * ftn,
     else if (tk == MscorlibBinder::GetMethod(METHOD__UNSAFE__BYREF_AS)->GetMemberDef())
     {
         // Return the argument that was passed in.
-        static const BYTE ilcode[] = { CEE_LDARG_0, CEE_RET };
+        static const BYTE ilcode[] = { CEE_LDARG_0, CEE_STLOC_0, CEE_LDLOC_0, CEE_RET };
```

Comment:
You would need to manufacture a signature for the local here. It can be done, but it is not a one liner. I think it is better to workaround it in C# in the Span implementation instead, like: https://github.com/dotnet/coreclr/pull/11212